fe/fe-core/src/main/cup/sql_parser.cup

```diff
@@ -6402,7 +6402,23 @@ non_pred_expr ::=
   | KW_NULL KW_IS KW_NOT KW_NULL
   {: RESULT = new BoolLiteral(false); :}
   | KW_CONVERT LPAREN expr:e COMMA type_def:targetType RPAREN
-  {: RESULT = new CastExpr(targetType, e); :}
+  {:
+    CastExpr castExpr = new CastExpr(targetType, e);
```

Comment:
I think this should be implemented as a SQL RewriteRule, better not doing any rewrite in parser.
And we also need to implement it in new query optimizer: Nereids.